### DISCUSSION TITLE
test djl just with java.exe in cmd resulting in UnsupportedOperationException

### DISCUSSION THREAD

**Discussion by ai25395** [deepjavalibrary/djl#2610] [Q&A]
For not having an IDE on my pc,I try djl in cmd like this:   **./java.exe -classpath 'E:\jdk-20.0.1\userlib\test1\api-0.22.1.jar;E:\jdk-20.0.1\userlib\test1\slf4j-api-2.0.7.jar' ../testscript/testNd.java**

And the file testNd.java contains:
![image](https://github.com/deepjavalibrary/djl/assets/52650634/60fb8986-0afb-4af4-8243-8ce950cd57e9)

But it turns out:
![image](https://github.com/deepjavalibrary/djl/assets/52650634/531fa089-5d21-4767-bff5-8dec5d498510)

If someone could please tell me how can I handle this?

**Comment by frankfliu** [comment]
You need include at least one of full deeplearning engine to performance NDArray operations:

- PyTorch
- TensorFlow
- MXNet

See: https://docs.djl.ai/master/docs/engine.html
